src/scripts/plugin.js

```diff
@@ -1,217 +1,24 @@
-      output: 'Hello world!',
-    },
-    octave: {
-      code: 'printf(\'Hello world!\')',
-      output: 'Hello world! Hello world!',
-    },
-    SageMath: {
-      code: 'print(\'Hello world!\')',
-      output: 'Hello world!',
-    },
-  };
-
-  if (sample[language] === undefined) return 'Error!';
-
-  return `
-    <label>Edit script:</label>
-    <pre data-executable="true" data-language=${language}>
-      ${sample[language].code}
-    </pre>
-    <div data-output="true">
-      ${sample[language].output}
-    </div>
-  `;
-};
-
-const changeAllLanguages = (editor, language = 'python3') => {
-  // changes the data-language attribute of all pre tags
-  const blockList = editor.document.getElementsByTag('pre');
-  blockList.toArray().forEach((codeBlock) => {
-    if (codeBlock.getAttribute('data-language')) {
-      codeBlock.setAttribute('data-language', language);
-    }
-  });
-};
-
-const insertWarning = () => `
-    <label class="warning"> Please do not click OK until code has finished executing. </label>`;
-
-
-const dialogConfig = (editor) => ({
-  title: 'Insert Interactive Script',
-  minHeight: 100,
-  minWidth: 400,
-  onShow() {
-    const dialog = this;
-    const language = getLanguage(editor);
-    dialog.setValueOf('tab-basic', 'language', language);
-    this.resize(500, 500);
-  },
-  onCancel() {
-    const dialog = this;
-    const editorLanguage = getLanguage(editor);
-    const dialogLanguage = dialog.getValueOf('tab-basic', 'language');
-    if (editorLanguage !== dialogLanguage) {
-      activateThebelab(thebelabConfig);
-    }
-  },
-  contents: [
-    {
-      id: 'tab-basic',
-      label: 'Basic Settings',
-      elements: [
-        {
-          type: 'select',
-          id: 'language',
-          label: 'Select language for page:',
-          items: [['Python 3'], ['Julia'], ['R'], ['Octave'], ['SageMath']],
-          default: 'Python 3',
-          onChange() {
-            const element = this.getDialog().getContentElement('tab-basic', 'code').getElement();
-            element.setHtml(editScriptAreaHTML(languageDictionary[this.getValue()]));
-            activateThebelab(thebelabConfig);
-          },
-        },
-        {
-          type: 'html',
-          id: 'code',
-          html: editScriptAreaHTML(),
-        },
-        {
-          type: 'hbox',
-          id: 'checkboxes',
-          widths: ['25%', '25%', '50%'],
-          children: [
-            {
-              type: 'checkbox',
-              id: 'no-output',
-              label: 'Insert without output',
-            },
-            {
-              type: 'checkbox',
-              id: 'no-code',
-              label: 'Insert with code hidden',
-            },
-            {
-              type: 'html',
-              html: '',
-            },
-          ],
-        },
-        {
-          type: 'html',
-          id: 'insert-warning',
-          html: insertWarning(),
-        },
-      ],
-    },
-  ],
-  onOk() {
```

Comment:
One bug: if you change the languages multiple times in succession within a short time span, and then click OK, it leads to a TypeError.
Changing the languages many times in succession is actually a universal bug since the user will most likely lose connection to the kernel. The TypeError error seems specific to this PR though.

Reply:
@celine168 I don't see a bug there. Can you produce a step by step way to reproduce this bug?

Reply:
Marking as resolved. Moving this to #46.